torsiondrive/qm_engine.py

```diff
@@ -327,6 +335,166 @@ def load_native_output(self, filename='output.dat'):
         m.build_topology()
         return m
 
+class EngineGaussian09(QMEngine):
+    def load_input(self, input_file):
+        """
+        !!!only Cartesian molecule specification is supported at the moment!!!
+        Load Gaussian09 input
+        Example input file:
+
+        %Mem=6GB
+        %NProcShared=2
+        %Chk=lig
+        # B3LYP/6-31G(d) Opt=ModRedundant
+
+        water energy
+
+        0   1
+        O  -0.464   0.177   0.0
+        H  -0.464   1.137   0.0
+        H   0.441  -0.143   0.0
+
+
+        """
+        elems, coords = [], []
+        reading_molecule, found_geo = False, False
+        gauss_temp = []  # store a template of the input file for generating new ones
+        with open(input_file) as gauss_in:
+            for line in gauss_in:
+                ls = line.split()
+                if len(ls) == 4 and check_all_float(ls[1:]):
+                    reading_molecule = True
+                    elems.append(ls[0])
+                    coords.append(ls[1:])
+                    if not found_geo:
+                        found_geo = True
+                        gauss_temp.append("$!geometry@here")
+
+                elif reading_molecule:
+                    if line.strip().lower() == '':
```

Comment:
`line.strip().lower()` can be replaced by `line.strip()`.
This means a single empty line will be needed after the xyz block, is that always true, and is it safe to insert the opt block at the first empty line following xyz block?

Reply:
Thanks for that change and yes that is always true there should be only one blank line to signal the end of the xyz input and then blank lines to signal the end of the input file.